cunumeric/deferred.py

```diff
@@ -483,6 +483,42 @@ def _copy_store(self, store: Any) -> DeferredArray:
         store_copy.copy(store_to_copy, deep=True)
         return cast(DeferredArray, store_copy)
 
+    def _slice_store(
```

Comment:
Can you make this into a `@staticmethod`, since it doesn't really use the `self` argument?